common/cpp/react/renderer/components/safeareacontext/RNCSafeAreaViewShadowNode.cpp

```diff
@@ -0,0 +1,107 @@
+  if (edgeValue.unit != YGUnitUndefined) {
+    return edgeValue;
+  }
+  YGValue axisValue = edges[axis];
+  if (axisValue.unit != YGUnitUndefined) {
+    return axisValue;
+  }
+  return edges[YGEdgeAll];
+}
+
+void RNCSafeAreaViewShadowNode::adjustLayoutWithState() {
+  ensureUnsealed();
+
+  auto props = getConcreteProps();
+  auto state =
+      std::static_pointer_cast<const RNCSafeAreaViewShadowNode::ConcreteState>(
+          getState());
+  auto stateData = state->getData();
+  auto edges = props.edges;
+
+  // Get the current values for padding / margin. The only caveat here is that
+  // percent values are not supported. Also might need to add support for start
+  // / end.
+  YGValue top, left, right, bottom;
+  if (props.mode == RNCSafeAreaViewMode::Padding) {
+    top = valueFromEdges(props.yogaStyle.padding(), YGEdgeTop, YGEdgeVertical);
+    left =
+        valueFromEdges(props.yogaStyle.padding(), YGEdgeLeft, YGEdgeHorizontal);
+    bottom =
+        valueFromEdges(props.yogaStyle.padding(), YGEdgeBottom, YGEdgeVertical);
+    right = valueFromEdges(
+        props.yogaStyle.padding(), YGEdgeRight, YGEdgeHorizontal);
+  } else {
+    top = valueFromEdges(props.yogaStyle.margin(), YGEdgeTop, YGEdgeVertical);
+    left =
+        valueFromEdges(props.yogaStyle.margin(), YGEdgeLeft, YGEdgeHorizontal);
+    bottom =
+        valueFromEdges(props.yogaStyle.margin(), YGEdgeBottom, YGEdgeVertical);
+    right =
+        valueFromEdges(props.yogaStyle.margin(), YGEdgeRight, YGEdgeHorizontal);
+  }
+  if (std::find(edges.begin(), edges.end(), "top") != edges.end()) {
+    top = yogaStyleValueFromFloat(
+        stateData.insets.top + (top.unit == YGUnitPoint ? top.value : 0));
+  }
+  if (std::find(edges.begin(), edges.end(), "left") != edges.end()) {
+    left = yogaStyleValueFromFloat(
+        stateData.insets.left + (left.unit == YGUnitPoint ? left.value : 0));
+  }
+  if (std::find(edges.begin(), edges.end(), "right") != edges.end()) {
+    right = yogaStyleValueFromFloat(
+        stateData.insets.right + (right.unit == YGUnitPoint ? right.value : 0));
+  }
+  if (std::find(edges.begin(), edges.end(), "bottom") != edges.end()) {
+    bottom = yogaStyleValueFromFloat(
+        stateData.insets.bottom +
+        (bottom.unit == YGUnitPoint ? bottom.value : 0));
+  }
+
+  YGStyle adjustedStyle = getConcreteProps().yogaStyle;
+  if (props.mode == RNCSafeAreaViewMode::Padding) {
+    adjustedStyle.padding()[YGEdgeTop] = top;
+    adjustedStyle.padding()[YGEdgeLeft] = left;
+    adjustedStyle.padding()[YGEdgeRight] = right;
+    adjustedStyle.padding()[YGEdgeBottom] = bottom;
+  } else {
+    adjustedStyle.margin()[YGEdgeTop] = top;
+    adjustedStyle.margin()[YGEdgeLeft] = left;
+    adjustedStyle.margin()[YGEdgeRight] = right;
+    adjustedStyle.margin()[YGEdgeBottom] = bottom;
+  }
+
+  auto currentStyle = yogaNode_.getStyle();
+  if (adjustedStyle.padding()[YGEdgeTop] != currentStyle.padding()[YGEdgeTop] ||
```

Comment:
Could we compare `adjustedStyle != currentStyle` instead of comparing every value independently?

Alternatively, compare specifically only padding and margin instead of writing out all edges.

Reply:
Comaping paddings / margin leads to this compiler error `Invalid operands to binary expression ('IdxRef<YGEdge, &YGStyle::padding_>' and 'IdxRef<YGEdge, &YGStyle::padding_>')`. I assume it doesn't implement the proper compare functions in yoga.